### DISCUSSION TITLE
How to debug completions not being provided

### DISCUSSION THREAD

**Discussion by jspawar** [joaotavora/eglot#740] [Q&A]
Using solargraph in a [private] project and wanted to debug lack of completions being provided.

I'm fairly certain that it should be able to provide completions at the point I'm editing in because:
1. it will _sometimes_ provide a completion suggestion once I've already type out like 80-90% of a thing I wanted completion for 
2. I'm doing it multiple different ways and locations in code (e.g. a new function, an existing function, in a proc, etc.)
3. it's a common and well-documented function (e.g. trying to complete on `Rails.logger.` which has multiple, well-documented methods or a method I've annotated myself with YARD doc comments)

Thanks y'all.

**Comment by mdbergmann** [comment]
I have a similar issue where there is a client request (`textDocument/completions`) and the server sends completions but they are not shown in eglot. What could be the reason?

```
[client-request] (id:20) Tue Feb 15 13:33:17 2022:
(:jsonrpc "2.0" :id 20 :method "textDocument/completion" :params
          (:textDocument
           (:uri "file:///foo.lfe")
           :position
           (:line 149 :character 6)
           :context
           (:triggerKind 2 :triggerCharacter ":")))
[server-reply] (id:20) Tue Feb 15 13:33:17 2022:
(:id 20 :result
     [(:label "columns/0" :kind 3 :detail "io:" :insertText "columns")
      (:label "columns/1" :kind 3 :detail "io:" :insertText "columns")
      (:label "format/1" :kind 3 :detail "io:" :insertText "format")])
```

**Reply by mdbergmann** [reply]
Sorry for reusing this question.
I've made an animated gif to show the issue.
On the open paren there are completions showing. I'm selecting the first one with 'enter'.
But on the colon `:` there should be a new set of completions showing, but they are not despite the 'completions' request/response.

![emacs-eglot-lfe-ls](https://user-images.githubusercontent.com/2143978/154080492-99635200-4641-40b2-ab72-4c0500b08dfb.gif)
